platform/osx/os_osx.mm

```diff
@@ -2270,12 +2270,12 @@ static int get_screen_index(NSScreen *screen) {
 Size2 OS_OSX::get_real_window_size() const {
 
 	NSRect frame = [window_object frame];
-	return Size2(frame.size.width, frame.size.height);
+	return Size2(frame.size.width, frame.size.height) * _display_scale();
 }
 
 void OS_OSX::set_window_size(const Size2 p_size) {
 
-	Size2 size = p_size;
+	Size2 size = p_size / _display_scale();
```

Comment:
I believe that `setFrame` will handle the `contentScaleFactor` of the `NSWindow`. The way this is written, if I'm on a retina display to get 1080, I must `set_window_size` with `3840, 2160` which seems unintuitive to me.

Reply:
`setFrame` do, but currently all other window position and size do this kind of scaling (only is project is configured to support hiDPI), we should either add it to these two or remove it everywhere else. Also, all internal measurements and input events use real pixel coordinates.